packages/medusa/src/services/fulfillment.ts

```diff
@@ -107,13 +121,16 @@ class FulfillmentService extends BaseService {
    * fulfillable quantity is lower than the requested fulfillment quantity.
    * Fulfillable quantity is calculated by subtracting the already fulfilled
    * quantity from the quantity that was originally purchased.
-   * @param {LineItem} item - the line item to check has sufficient fulfillable
+   * @param item - the line item to check has sufficient fulfillable
    *   quantity.
-   * @param {number} quantity - the quantity that is requested to be fulfilled.
-   * @return {LineItem} a line item that has the requested fulfillment quantity
+   * @param quantity - the quantity that is requested to be fulfilled.
+   * @return a line item that has the requested fulfillment quantity
    *   set.
    */
-  validateFulfillmentLineItem_(item, quantity) {
+  validateFulfillmentLineItem_(
+    item: LineItem | undefined,
+    quantity: number
+  ): Omit<LineItem, "beforeInsert"> | null {
```

Comment:
suggestion: Instead of having to omit some properties (even if having those private methods in the types is weird 😂 ) could we do something like `return LineItemRepo.create({ /* ...data */ })`

Reply:
I have done the lineItemRepo instead. Alternative would be a `partial<lineitem>`. I have also removed the transformer function from the `getFulfillmentItems_` method and instead just reference the `validateFulfillmentLineItem_` function directly since the method is not used elsewhere. 

@olivermrbl wdyt?